src/main/java/com/google/songgame/servlets/GameServlet.java

```diff
@@ -46,21 +50,122 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(roundEntity);
   }
-  
 
-  private String getAuthToken() {
-    String clientId = "1c29ff191b444611a6d9dbb4a354642f";
-    String clientSecret = "6b92b6eb8fe24d5e98178407d3099bd8";
-    SpotifyApi spotifyApi = new SpotifyApi.Builder().setClientId(clientId).setClientSecret(clientSecret).build();
-    ClientCredentialsRequest clientCredentialsRequest = spotifyApi.clientCredentials().build();
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String playlistUrl = getParameter(request, "playlist-link", "");
+    setvideoId(playlistUrl);
+    response.sendRedirect("/game.html");
+  }
 
+  /**
+   * Returns the request parameter, or the default value if the parameter was not specified by the
+   *     client
+   */
+  private String getParameter(HttpServletRequest request, String name, String defaultValue) {
+    String value = request.getParameter(name);
+    if (value == null) {
+      return defaultValue;
+    }
+    return value;
+  }
+
+  /**
+   * Use playlist url to connect to YouTube API and set video ID
+   *
+   */
+  private void setvideoId(String playlistUrl) {
+    String playlistId = getIdFromUrl(playlistUrl);
+    PlaylistItemListResponse playlistItem = new PlaylistItemListResponse();
+    // Retrieve Playlist item from Youtube API
     try {
-      ClientCredentials clientCredentials = clientCredentialsRequest.execute();
-      return clientCredentials.getAccessToken();
-    } catch (IOException | SpotifyWebApiException | ParseException e) {
-      System.out.println("Error: " + e.getMessage());
+      playlistItem = getPlaylistInfo(playlistId);
+    } catch (Exception e) {
+      e.printStackTrace();
+    }
+    // Parse Playlist item Json string to retrieve video IDs
+    String playlistItemJson = new Gson().toJson(playlistItem);
+    ArrayList<String> playlistVideos = parsePlaylistItem(playlistItemJson);
+    playlistSize = playlistVideos.size();
```

Comment:
I don't think it makes sense to set an instance variable here, but then use it the getRandomVideo private method.

I think it might be better to derive the size of the playlistVideos in that method.